src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs

```diff
@@ -239,34 +239,19 @@ internal static PageBoundPropertyDescriptor[] CreateBoundProperties(TypeInfo typ
         {
             var properties = PropertyHelper.GetVisibleProperties(type.AsType());
 
-            // If the type has a [BindPropertyAttribute] then we'll consider any and all public properties bindable.
-            var bindPropertyOnType = type.GetCustomAttribute<BindPropertyAttribute>();
-
             var results = new List<PageBoundPropertyDescriptor>();
             for (var i = 0; i < properties.Length; i++)
             {
                 var property = properties[i];
                 var bindingInfo = BindingInfo.GetBindingInfo(property.Property.GetCustomAttributes());
 
                 // If there's no binding info then that means there are no model binding attributes on the
-                // property. So we won't bind this property unless there's a [BindProperty] on the type.
-                if (bindingInfo == null && bindPropertyOnType == null)
+                // property. So we won't bind this property.
+                if (bindingInfo == null)
                 {
                     continue;
                 }
 
-                // If this property is declared as part of a pages base class, then it's likely infrastructure,
-                // so skip it.
-                if (property.Property.DeclaringType.GetTypeInfo().IsDefined(typeof(PagesBaseClassAttribute)))
```

Comment:
Will do!

Yeah we don't need that anymore. It's a complicated thing to explain and it made me a little uncomfortable in the first place. So I'm glad to be rid of it.